std-bits/base/src/main/java/org/enso/base/Stream_Utils.java

```diff
@@ -2,9 +2,84 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.OutputStream;
 import java.util.Arrays;
 
-public class Stream_Utils {
+public final class Stream_Utils {
+  private Stream_Utils() {}
+
+  /**
+   * Conversion interface. Any Enso/Truffle object with invocable {@read} member that takes three
+   * arguments is eligible for being treated as host Java {@link InputStream}. There are two
+   * overloaded {@link #asInputStream conversion methods}. The <em>hosted Java interop</em> system
+   * of Truffle will pick the more suitable one depending on the type of argument.
+   *
+   * @see #asInputStream
+   */
+  public static interface InputStreamLike {
+    public int read(byte[] arr, int off, int len) throws IOException;
+
+    public default int available() {
+      return 0;
```

Comment:
When would this implementation of `available` be used?

Reply:
The idea was:
- it is enough to offer invocable `read` member to be `InputStreamLike`
- offering  invocable `available` member  is optional
- if the `available` member isn't provided, then `available` returns `0`

I believe it really works that way, but I haven't tested it. True.

Reply:
Test written in bfa60b4 with the help of `foreign js` implementation. Thanks for encouraging me to increase test coverage.